src/main/java/org/jabref/gui/openoffice/ManageCitationsDialogViewModel.java

```diff
@@ -31,33 +33,37 @@
     private final OOBibBase ooBase;
     private final DialogService dialogService;
 
-    public ManageCitationsDialogViewModel(OOBibBase ooBase, DialogService dialogService) throws NoSuchElementException, WrappedTargetException, UnknownPropertyException {
+    public ManageCitationsDialogViewModel(OOBibBase ooBase, DialogService dialogService)
+        throws
+        NoSuchElementException,
+        WrappedTargetException,
+        UnknownPropertyException,
```

Comment:
Are these needed?

Reply:
Do you mean the line breaks, or the exceptions?
The line breaks technically no.

Some of these exceptions mentioned here I caught below,
(UnknownPropertyException, WrappedTargetException)
but not all. 

The contructor itself is called from `ManageCitationsDialogView.initialize()`

```
    @FXML
    private void initialize() throws NoSuchElementException, WrappedTargetException, UnknownPropertyException, JabRefException {
```

There is no visible call to initialize() in the class body, presumably accessed through the `@FXML` magic,
I guess it is called as an event handler or action.

I have no definit answer to "where should these exceptions be caught": should we catch them all
here, or pass them all up (to initialize? or above that?), or something in between. We are probably choosing between 
calling a dialog box from an event handler and throwing exception from one. 

Those I caught were probably caught here, because this is where they reached
the GUI level. I may have seen UnknownPropertyException and caught it here,
I am not sure I have ever seen a WrappedTargetException. Maybe.

Summary: I do not know where should they be caught.

Moved them to the catch below, then dropped NoSuchElementException,
since the compiler now noticed it is not thrown anyway.

Reply:
> Do you mean the line breaks, or the exceptions?

Exceptions.

> There is no visible call to initialize() in the class body, presumably accessed through the @FXML magic,
I guess it is called as an event handler or action.

I haven't spent that much time with JavaFX, but [pretty much](https://openjfx.io/javadoc/16/javafx.fxml/javafx/fxml/doc-files/introduction_to_fxml.html#controllers)

Reply:
> I have no definit answer to "where should these exceptions be caught": should we catch them all
here, or pass them all up (to initialize? or above that?), or something in between. We are probably choosing between
calling a dialog box from an event handler and throwing exception from one.

I'd say catch them all here, or latest in the View. If appropriate do some action (although in this case, I am not sure if you can do all that much more than display an error dialog to the user). I'll try to update the documentation when I have time.

I'll comment in the code what I think of each and why but quite frankly, some of them I have no idea what to do with either (`WrappedTargetException` would be one of those). 

---

As a general rule can't think of any reason to make a `JabRefException` (or any other) bubble up past the View. I'd say that most of them (all?) should be handled in the ViewModel. This would be one of those I'd appreciate a second opinion on @Siedlerchr 😛